contacts_app/lib/contactsPage.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:flutter/material.dart';
+import 'package:contacts_service/contacts_service.dart';
+
+class ContactsPage extends StatefulWidget {
+  @override
+  _ContactsPageState createState() => _ContactsPageState();
+}
+
+class _ContactsPageState extends State<ContactsPage> {
+  Iterable<Contact> _contacts;
+
+  @override
+  void initState() {
+    getContacts();
+    super.initState();
+  }
+
+  Future<void> getContacts() async {
+    //We already have permissions for contact when we get to this page, so we
+    // are now just retrieving it
+    final Iterable<Contact> contacts = await ContactsService.getContacts();
+    setState(() {
+      _contacts = contacts;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: (Text('Contacts')),
+      ),
+      body: _contacts != null
+          //Build a list view of all contacts, displaying their avatar and
+          // display name
+          ? ListView.builder(
+              itemCount: _contacts?.length ?? 0,
+              itemBuilder: (BuildContext context, int index) {
+                Contact contact = _contacts?.elementAt(index);
```

Comment:
what happens if contacts is null?

Reply:
A null contact can't be returned. But if the contacts fields are null (using display name and avatar in this case), null check is applied below